### PR TITLE
bsunanda:Run2-TB10 correct beam line description of CERN HGCal test beam simulation

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: SimG4CMS/HGCalTestBeam/plugins/HGCalTBAnalyzer.cc [hunk 1]
```diff
@@ -560,14 +560,14 @@ void HGCalTBAnalyzer::analyzeSimHits (int type, std::vector<PCaloHit>& hits) {
 #endif
     hSimHitLng1_[type]->Fill(layer,energy);
     if (type == 0) {
-      if (layer<(int)(hSimHitLayEn2E_.size())) {
-	simHitLayEn2E[layer] = energy;
-	hSimHitLayEn2E_[layer]->Fill(energy);
+      if (layer-1 < (int)(hSimHitLayEn2E_.size())) {
```
Comment (davidlange6) [inline SimG4CMS/HGCalTestBeam/plugins/HGCalTBAnalyzer.cc]: note that layer is never used aside from layer-1 - perhaps wise to define layer= itr->first -1; ? @bsunanda

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch It needs your signature again - modified to take care of David's comments

**Comment by bsunanda** [comment]
@civanch @ianna Could you please re-approve this after making changes suggested by David

**Comment by ianna** [comment]
+1

**Pull request merged**